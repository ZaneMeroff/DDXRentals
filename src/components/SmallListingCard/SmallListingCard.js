import React, { Component } from 'react';
import './SmallListingCard.scss';
import snowflake from '../../data/snowflake.png';
import compassIcon from '../../data/compass_icon.png';
import gondolaIcon from '../../data/gondola_icon.png';
import priceIcon from '../../data/price_icon.png';
import featuresIcon from '../../data/features_icon.png';

class SmallListingCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      cardExpand: false
    }
  }

  expandCard = () => {
    this.setState({cardExpand: !this.state.cardExpand})
  }

  allFeatures = () => this.props.features.map(feature => {
    return <li>{feature}</li>
  })

  render() {

    let cardState = this.state.cardExpand ? '' : 'expand';

    return (
    <div className='article-list'>
      <article className='short-card-container'>
        <img src={process.env.PUBLIC_URL + `/images/${this.props.id}_a.jpg`} alt='Listing' className='image-container'/>
        <div className='small-card-details'>
          <h3 id='location-name'><img className='icon' src={gondolaIcon} alt='location icon'/>{this.props.name}</h3>
          <h3><img className='icon' src={compassIcon} alt='location icon'/>{this.props.address} {this.props.addressZip}</h3>
          <h3><img className='icon' src={priceIcon} alt='location icon'/>{this.props.cost} per night</h3>
        </div>
        <div className='card-buttons'>
          <img type='button' className='fav-icon' src={snowflake} alt='snowflake'/>
          <button className='view-listing-button' onClick={this.expandCard.bind(this)}>view details</button>
        </div>
      </article>

      <article className={`full-card-container ${cardState}`}>
        <div className='additional-images-container'>
          <img src={process.env.PUBLIC_URL + `/images/${this.props.id}_b.jpg`} alt='Listing' className='additional-images'/>
          <img src={process.env.PUBLIC_URL + `/images/${this.props.id}_c.jpg`} alt='Listing' className='additional-images'/>
        </div>
        <div className='additional-info'>
          <h3>Beds: {this.props.beds}</h3>
          <h3>Baths: {this.props.baths}</h3>
          <h3><img className='icon' src={featuresIcon} alt='location icon'/>features:</h3>
          <ul>
            {this.allFeatures}
          </ul>
        </div>
        <button className='make-reservation-button'>make reservation</button>
      </article>

    </div>
  )

}

}

export default SmallListingCard;
