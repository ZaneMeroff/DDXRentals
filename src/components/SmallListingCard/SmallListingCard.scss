
.short-card-container,
.full-card-container {
  align-items: center;
  background-color: rgba(255, 255, 255, 0.8);
  border-radius: 35px;
  display: flex;
  flex-direction: row;
  height: auto;
  justify-content: space-around;
  width: 750px;
  z-index: 100;
}

.short-card-container {
  margin-top: 6%;
  border-bottom-left-radius: 0px;
  border-bottom-right-radius: 0px;
}

.full-card-container {
  visibility: hidden;
  margin-bottom: 3%;
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
}

h3 {
  font-size: 1.3em;
  margin: 3% 0% 3% 0%;
}

.image-container,
.additional-images {
  background-color: #FFBF03;
  border-radius: 20px;
  height: 150px;
  width: 300px;
  margin: 20px;
  object-fit: cover;
  // max-height: 150px;
  // max-width: 300px;
}

.additional-images {
  // margin: 16% 0% 16% 0%;
}

.icon {
  height: 30px;
  margin-right: 10px;
  width: auto;
}

.small-card-details,
.additional-info {
  align-items: flex-start;
  display: flex;
  flex-direction: column;
  width: 320px;
}

ul {
  margin-left: 15%;
}

li {
  margin: 5%;
  width: 100%;
  font-size: 1.2em;
}

.additional-info {
  align-self: flex-start;
  // margin-top: -3%;
}

.fav-icon {
  height: 70px;
  margin-bottom: 30%;
  width: 70px;
}

.card-buttons {
  align-items: center;
  display: flex;
  flex-direction: column;
}

.view-listing-button,
.make-reservation-button {
  background-color: #FFBF03;
  border-radius: 15px;
  color: #fff;
  font-size: 1em;
  height: 50px;
  width: 120px;
}

.make-reservation-button {
  margin-top: -40%;
}

.view-listing-button:hover,
.make-reservation-button:hover,
.fav-icon:hover,
#form-btn {
  cursor: pointer;
  transform: scale(1.1);
  transition: transform .5s;
  background-color: #fcd150;
}

.additional-images-container {
  display: flex;
  flex-direction: column;
  // border: 2px solid red;
}

.expand {
  visibility: visible;
  .short-card-container {
    border-bottom-left-radius: 0px;
    border-bottom-right-radius: 0px;
  }
}

#location-name {
  font-size: 2em;
  color: #DB49CB;
}
