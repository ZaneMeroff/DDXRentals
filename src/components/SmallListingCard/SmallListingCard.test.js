import React from 'react';
import ReactDOM from 'react-dom';
import SmallListingCard from './SmallListingCard';
import { shallow } from 'enzyme';

describe('SmallListingCard', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SmallListingCard />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should be an instance of SmallListingCard component', () => {
    const wrapper = shallow(<SmallListingCard

      />);
      expect(wrapper).toMatchSnapshot();
  });

});
