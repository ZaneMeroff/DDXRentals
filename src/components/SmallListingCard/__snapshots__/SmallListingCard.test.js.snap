// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallListingCard should be an instance of SmallListingCard component 1`] = `
<div
  className="article-list"
>
  <article
    className="short-card-container"
  >
    <img
      alt="Listing"
      className="image-container"
      src="/images/3_a.jpg"
    />
    <div
      className="small-card-details"
    >
      <h3
        id="location-name"
      >
        <img
          alt="location icon"
          className="icon"
          src="gondola_icon.png"
        />
        Hip RiNo Party Spot
      </h3>
      <h3>
        <img
          alt="location icon"
          className="icon"
          src="compass_icon.png"
        />
        2250 Lawrence St
         
        80205
      </h3>
      <h3>
        <img
          alt="location icon"
          className="icon"
          src="price_icon.png"
        />
        420
         per night
      </h3>
    </div>
    <div
      className="card-buttons"
    >
      <img
        alt="snowflake"
        className="fav-icon"
        onClick={[Function]}
        src="snowflake.png"
        type="button"
      />
      <button
        className="view-listing-button"
        onClick={[Function]}
      >
        view details
      </button>
    </div>
  </article>
  <article
    className="full-card-container expand"
  >
    <div
      className="additional-images-container"
    >
      <img
        alt="Listing"
        className="additional-images"
        src="/images/3_b.jpg"
      />
      <img
        alt="Listing"
        className="additional-images"
        src="/images/3_c.jpg"
      />
    </div>
    <div
      className="additional-info"
    >
      <h3>
        Beds: 
        3
      </h3>
      <h3>
        Baths: 
        2.5
      </h3>
      <h3>
        <img
          alt="location icon"
          className="icon"
          src="features_icon.png"
        />
        features:
      </h3>
      <ul>
        <li>
          hot tub
        </li>
        <li>
          espresso machine
        </li>
      </ul>
    </div>
    <button
      className="make-reservation-button"
    >
      make reservation
    </button>
  </article>
</div>
`;
